service/routes.py

```diff
@@ -174,14 +174,7 @@ def delete_promotions(promotion_id):
 ######################################################################
 def check_content_type(content_type) -> None:
     """Checks that the media type is correct"""
-    if "Content-Type" not in request.headers:
-        app.logger.error("No Content-Type specified.")
-        abort(
-            status.HTTP_415_UNSUPPORTED_MEDIA_TYPE,
-            f"Content-Type must be {content_type}",
-        )
-
-    if request.headers["Content-Type"] == content_type:
```

Comment:
Why are we removing the error logging here? Should we not keep this bit?

Reply:
+1

Reply:
With my changes, the `if "Content-Type" not in request.headers:` case will go to line 180, to log Invalid Content-Type error, I do this change because if we do not include "Content-Type" in the request header, this request will then have status code 201 rather than HTTP_415_UNSUPPORTED_MEDIA_TYPE that we expect.

Reply:
That is the Flask behavior when handling the `Content-Type` header during HTTP requests. For instance, consider the following example:

```python
def test_create_promotion_without_content_type_header(self):
    """It should raise a 415 error for missing Content-Type header"""
    test_promotion = PromotionFactory()
    response = self.client.post(
        BASE_URL,
        # json=test_promotion.serialize(),
    )
    self.assertEqual(response.status_code, status.HTTP_415_UNSUPPORTED_MEDIA_TYPE)
    error_message = response.get_json()
    self.assertIn("Content-Type must be", error_message["message"])
```

In this case, When the `json` parameter is provided in the request. Flask automatically sets the `Content-Type` to `application/json` even without explicit header setting. However, if we comment out that line like the above. Then no `Content-Type` header is set and it will raise the error 415 Unsupported Media Type as we expect.